contractcourt/channel_arbitrator.go

```diff
@@ -412,7 +412,8 @@ func (c *ChannelArbitrator) Start() error {
 	if startingState == StateWaitingFullResolution &&
 		nextState == StateWaitingFullResolution {
 
-		if err := c.relaunchResolvers(); err != nil {
+		confCommitSet := commitSet.HtlcSets[*commitSet.ConfCommitKey]
```

Comment:
Can you add a comment here explaining why this line can never be hit with a `nil` commit set? (`errNoCommitSet`)

Reply:
When we decide which resolvers to create, we also look at other versions of the commit tx. Could it somehow be that a resolver was created that needs info from a different commit tx version? I seems unlikely, because we mostly create `HtlcFailNowAction` in those cases, but checking just in case.

Reply:
> Could it somehow be that a resolver was created that needs info from a different commit tx version?

Off the top, I can't think of any such scenario. Resolvers are created based on what's in the `htlcActions` map. The `htlcActions` map[populated here](https://github.com/lightningnetwork/lnd/blob/master/contractcourt/channel_arbitrator.go#L1605) by the `constructChainActions` method which only uses the `confCommitSet`. The `confCommitSet` is written to disk as soon as we receive an event of the channel closing on chain. 

If we want to play it safe, we can very well just include _every_ HTLC outpoint in the map to supplement all resolvers.

Reply:
For me, I'd rather not play safe if we don't think it will happen. Can only confuse devs looking at it later.

I was thinking about `checkLocalChainActions` called in `StateDefault` using `activeHtlcs` coming from the link which may contain htlcs that are not available in the finally confirmed commit set. But I don't think that can happen, because the chain actions are only used to decide whether to go on chain at that point and discarded.